documentation/private-link-service/private-link-redirect-map.md

```diff
@@ -0,0 +1,170 @@
+# DASH Private Link Redirect Map HLD
+| Rev | Date | Author | Change Description |
+| --- | ---- | ------ | ------------------ |
+| 0.1 | 01/26/2025 | Riff Jiang, Junhua Zhai| Initial version |
+
+1. [Terminology](#1-terminology)
+2. [Background](#2-background)
+3. [SDN transformation](#3-sdn-transformation)
+    - 3.1 [Redirect map ](#31-redirect-map)
+    - 3.2 [Redirect map with PL NSG](#32-redirect-map-with-pl-nsg)
+    - 3.3 [Redirect map with fast path](#33-redirect-map-with-fast-path)
+    - 3.4 [Non-required features](#34-non-required-features) 
+4. [Resource modeling, requirement, and SLA](#4-resource-modeling-requirement-and-sla)
+    - 4.1 [Scaling requirement](#41-scaling-requirement)
+    - 4.2 [Reliability requirements](#42-reliability-requirements)
+5. [SAI API design](#5-sai-api-design)
+    - 5.1 [DASH Outbound Port Map](#51-dash-outbound-port-map)
+    - 5.2 [DASH Outbound Port Map Port Range](#52-dash-outbound-port-map-port-range)
+    - 5.3 [DASH CA-PA mapping attributes](#53-dash-ca-pa-mapping-attributes)
+6. [DASH pipeline behavior](#6-dash-pipeline-behavior)
+    - 6.1 [VM-to-PLS direction](#61--vm-to-pls-direction-outbound)
+    - 6.2 [PLS-to-VM direction](#62-pls-to-vm-direction)
+
+## 1. Terminology
+| Term | Explanation |
+| ---  | --- |
+| ER   | [ExpressRoute]( https://learn.microsoft.com/en-gb/azure/expressroute/expressroute-introduction)|
+| MSEE | Microsoft Enterprise Edge |
+| CE   | Customer Edge |
+| MSEE/CE routers | Microsoft Enterprise Edge / Customer Edge routers. CE and MSEE routers are directly connected and forming pairs. |
+| PL   | [PrivateLink](https://azure.microsoft.com/en-us/products/private-link) |
+| PL NSG | PrivateLink NSG. NSG here is not VNET NSG, but a set of appliances that sits between PE and ER for security checks or statistics collection. |
+| PE   | Private endpoint |
+| PLS  | PrivateLink Service. This is the term for private endpoint from server side. Customer can create their private link service, then expose them to their VNETs as a private endpoint. |
+| AZ   | Availability zone |
+
+## 2. Background
+Azure Private Link provides private connectivity from a virtual network to Azure platform as a service, by providing an 1-to-1 VNET mapping to the service. To extend the mapping, PL redirect map adds port-level rewrite info that maps a single VNET IP to a list of services.
+
+## 3. SDN transformation
+### 3.1 Redirect map 
+Private Link redirect map is built on top of private link, using exactly the same sdn transformation as private link defined [here](https://github.com/sonic-net/DASH/blob/main/documentation/general/sdn-pipeline-basic-elements.md#private-link).
+
+The only difference is that the IP address used in underlay routing and 4to6 action will be specified by a port-based service mapping – for a specified port range, different IP or destination port can be used for crafting the packet.
+
+![PL Redirect VM-to-PLS direction](images/private-link-redirect-vm-to-pls.svg)
```

Comment:
will be better to export svg using this setting:

![image](https://github.com/user-attachments/assets/9979eefc-443f-49ba-9007-7fb8dab60647)

Reply:
Fixed